server/models/remote_provider.go

```diff
@@ -3024,34 +3024,32 @@ func (l *RemoteProvider) ExtensionProxy(req *http.Request) ([]byte, error) {
 	// then attach the final path to the remote provider URL
 	remoteProviderURL, _ := url.Parse(fmt.Sprintf("%s%s", l.RemoteProviderURL, path))
 	logrus.Debugf("constructed url: %s", remoteProviderURL.String())
+	
+	defer req.Body.Close()
+	bd, _ := io.ReadAll(req.Body)
+	bf := bytes.NewBuffer(bd)
 
 	// make http.Request type variable with the constructed URL
-	cReq, _ := http.NewRequest(req.Method, remoteProviderURL.String(), req.Body)
-	tokenString, err := l.GetToken(req)
-
+	cReq, err := http.NewRequest(req.Method, remoteProviderURL.String(), bf)
```

Comment:
Do you know why body was not passed correctly?
I think the issue was only bcz of incorrect headers.

Reply:
Hmm, I'm not sure. While debugging I realised that an empty body was being sent in the request to Meshery Cloud. So on other side there were sometimes error related to unmarshalling as the body itself was empty
So to take care of any edge case, I kind of added this extra step

_But we have been using these user extensions for a while and haven't seen that error. 🤔_

Reply:
After fixing headers, did you tried without reading and sending body explicitly?

Reply:
Yeah, now it seems to work fine without the conversion of the original body. I might have bad build while debugging.
So I reverted that change.